packages/core/admin/utils/plugins.js

```diff
@@ -0,0 +1,217 @@
+   *
+   * If the file doesn't exist well then it's probably a node_module, so instead we use `require.resolve`
+   * which will resolve the path to the module in node_modules. If it fails with the specific code `MODULE_NOT_FOUND`
+   * then it doesn't have an admin part to the package.
+   *
+   * NOTE: we should try to move to `./package.json[exports]` map with bundling of our own plugins,
+   * because these entry files are written in commonjs restricting features e.g. tree-shaking.
+   */
+  try {
+    const isLocalPluginWithLegacyAdminFile = fs.existsSync(
+      path.resolve(`${plugin.pathToPlugin}/strapi-admin.js`)
+    );
+
+    if (!isLocalPluginWithLegacyAdminFile) {
+      const isModuleWithFE = require.resolve(`${plugin.pathToPlugin}/strapi-admin`);
+
+      return isModuleWithFE;
+    }
+
+    return isLocalPluginWithLegacyAdminFile;
+  } catch (err) {
+    if (err.code === 'MODULE_NOT_FOUND') {
+      /**
+       * the plugin does not contain FE code, so we
+       * don't want to import it anyway
+       */
+      return false;
+    }
+
+    throw err;
+  }
+};
+
+/**
+ *
+ * @param {Plugin[]} plugins
+ * @param {string} dest
+ * @returns {void}
+ */
+async function createPluginFile(plugins, dest) {
+  const pluginsArray = plugins.map(({ pathToPlugin, name, info }) => {
+    const shortName = camelCase(name);
+
+    let realPath = '';
+
+    /**
+     * We're using a module here so we want to keep using the module resolution procedure.
+     */
+    if (info?.packageName || info?.required) {
+      /**
+       * path.join, on windows, it uses backslashes to resolve path.
+       * The problem is that Webpack does not windows paths
+       * With this tool, we need to rely on "/" and not "\".
+       * This is the reason why '..\\..\\..\\node_modules\\@strapi\\plugin-content-type-builder/strapi-admin.js' was not working.
+       * The regexp at line 105 aims to replace the windows backslashes by standard slash so that webpack can deal with them.
+       * Backslash looks to work only for absolute paths with webpack => https://webpack.js.org/concepts/module-resolution/#absolute-paths
+       */
+      realPath = path.join(pathToPlugin, 'strapi-admin').replace(/\\/g, '/');
+    } else {
+      realPath = path
+        .join(path.relative(path.resolve(dest, 'admin', 'src'), pathToPlugin), 'strapi-admin')
+        .replace(/\\/g, '/');
+    }
+
+    return {
+      name,
+      pathToPlugin: realPath,
+      shortName,
+    };
+  });
+
+  const content = `
+${pluginsArray
+  .map(({ pathToPlugin, shortName }) => {
+    const req = `'${pathToPlugin}'`;
+
+    return `import ${shortName} from ${req};`;
+  })
+  .join('\n')}
+
+
+const plugins = {
+${[...pluginsArray]
+  .map(({ name, shortName }) => {
+    return `  '${name}': ${shortName},`;
+  })
+  .join('\n')}
+};
+
+export default plugins;
+`;
+
+  return asyncFs.writeFile(path.resolve(dest, 'admin', 'src', 'plugins.js'), content);
```

Comment:
at the moment yes, because it's in the admin build package, i think when we move the bundling stuff to the CLI we won't need a custom file to be made :)